### PR TITLE
pass keystone admin password to neutron-ha-tool via file (bsc#922751)

### PR DESCRIPTION
**Pull request by aspiers**
Will need backporting to tex and maybe stoney too.

Requires corresponding change in neutron-ha-tool:

https://build.suse.de/request/show/57156

### CODE REVIEW

**Comment by aspiers** [comment]
Tested and it works, but haven't tested the resulting file with neutron-ha-tool yet.

File: chef/cookbooks/neutron/recipes/network_agents_ha.rb [hunk 1]
```diff
@@ -125,13 +125,22 @@
...
+  action :create
+end
+
 pacemaker_primitive ha_tool_primitive_name do
   agent node[:neutron][:ha][:network][:ha_tool_ra]
   params ({
     "os_auth_url"    => keystone_settings["internal_auth_url"],
     "os_tenant_name" => keystone_settings["admin_tenant"],
     "os_username"    => keystone_settings["admin_user"],
-    "os_password"    => keystone_settings["admin_password"],
```
Comment (vuntz) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: Looks good to me, but shouldn't there be a way in the OCF to specify the path to the file?

Comment (vuntz) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: Reason I'm asking is that I'm unsure I'd put it in /etc/neutron/os_password -- I'd probably make it clearer that it's a file for the ha tool...

Comment (aspiers) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: Well actually in the long term it's not necessarily just for the HA tool; it could be for anything which needs to communicate with the API - possibly even stuff outside neutron, e.g. the keystone OCF RA.  So you *could* argue it should be in `/etc/keystone`.  But right now it really shouldn't matter at all.

I can't imagine any reason why it would need to be a configurable parameter, so I suggest we stick with the simplest thing which could possibly work for now.  If we need to make it configurable later then we can easily do that, but that seems like a waste of effort right now.

**Comment by jsuchome** [comment]
This works as planned +1

**Comment by aspiers** [comment]
Added a patch for region name.

File: chef/cookbooks/neutron/recipes/network_agents_ha.rb [hunk 2]
```diff
@@ -125,13 +125,23 @@
...
+end
+
 pacemaker_primitive ha_tool_primitive_name do
   agent node[:neutron][:ha][:network][:ha_tool_ra]
   params ({
     "os_auth_url"    => keystone_settings["internal_auth_url"],
+    "os_region_name" => keystone_settings["endpoint_region"],
     "os_tenant_name" => keystone_settings["admin_tenant"],
     "os_username"    => keystone_settings["admin_user"],
-    "os_password"    => keystone_settings["admin_password"],
```
Comment (nkrinner) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: This breaks the code for me and I don't understand why this change is needed.

Comment (aspiers) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: It's explained in the commit message. Storing the password on disk is the whole point of this PR.

Comment (aspiers) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: It probably breaks because you don't have the patched neutron-ha-tool.  Strongly recommend you read the context before further testing.

**Comment by dirkmueller** [comment]
+1
